tests/hawkey/test_sack.cpp

```diff
@@ -83,14 +83,14 @@ START_TEST(test_list_arches)
     const char ** arches = dnf_sack_list_arches(sack);
 
     /* noarch, x86_64, athlon, i686, i586, i486, i386 */
-    fail_unless(g_strv_length((gchar**)arches) == 7);
+    fail_unless(6 <= g_strv_length((gchar**)arches));
 
-    if (strcmp(arches[2], "athlon") == 0) {
+    if (g_strv_length((gchar**)arches) == 7) {
         // Fedora, Mageia
-        ck_assert_str_eq(arches[3], "i686");
+        ck_assert_str_eq(arches[2], "athlon");
     } else {
-        // openSUSE - "athlon" is not available
-        ck_assert_str_eq(arches[3], "i586");
+        // openSUSE, Debian - "athlon" is not available
+        ck_assert_str_eq(arches[2], "i686");
```

Comment:
I know this goes beyond what you want to fix, but could you add asserts for all the items in the array, in both cases? The one item that's check is arbitrary and we know what should all the other values be, so we should check them all.

Reply:
I've added all the assert on each arch. I've tested build only under Debian at this moment.